docs/reference/yaml-format-reference.md

```diff
@@ -75,6 +82,27 @@ If a recipe generates files during the `generate` phase, those files will always
 To conditionally generate files, implement a custom scanning recipe. You can define logic in the scanning phase based on existing source files, and use that context in the generate phase to control whether a file should be created.
 :::
 
+#### Preconditions vs. Exclusions
+
+Since preconditions work on already-parsed files, they're not the right tool if you need to skip parsing certain files. Below is a table that should help you make an informed decision on when to use one feature over another:
+
+| Feature           | Purpose                     | When It Runs         | Use Case                                                              |
+|-------------------|-----------------------------|----------------------|-----------------------------------------------------------------------|
+| **Exclusions**    | Skip parsing files entirely | During parsing phase | Avoid parse errors, skip expensive directories, ignore generated code |
+| **PlainTextMask** | Parse files as plain text   | During parsing phase | Parse files that would fail language-specific parsers                 |
```

Comment:
We _could_ already not mention `PlainTextMask`, as we might drop it in the future.
```suggestion
```